### PR TITLE
Remove `MOVEIT_LIB_NAME`

### LINKED ISSUE #1720: Replace `MOVEIT_LIB_NAME` with target names
### Description

The `MOVEIT_LIB_NAME` CMake variable creates an extra layer of indirection when writing target-based CMake commands. This practice does not aid readability while harming the grep-ability of the build scripts. The variable ends up getting re-used for multiple targets which further harms ones' ability to understand the build scripts.

I propose we simply remove all instances of that variable and instead spell out the target name where it's being used. 

I'm not positive where this pattern came from but I believe it originated in the ROS 1 documentation and has been copied ever since them. I can't think of any compelling reasons to keep doing this. I've never heard any CMake experts recommend doing this nor do I think Kitware themselves recommend it.

### PR DESCRIPTION
**Pull request by ChrisThrasher**
### Description

Closes #1720

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by tylerjw — APPROVED** [review]
This is a nice cleanup, thank you.

**Review by henningkayser — APPROVED** [review]

**Comment by ChrisThrasher** [comment]
Do I need to rebase on `main` before this can be merged? I'm not sure if that's a requirement or not.

**Pull request merged**
